### DISCUSSION TITLE
Directly Display Panel

### DISCUSSION THREAD

**Discussion by jcummings2** [dannagle/PacketSender#301] [Q&A] [answered]
Is there a way to directly launch a panel? That is, I would like a specified panel to come up without the user having to click anything else. Via the command line is fine.

**Comment by dannagle** [comment, marked as answer]
Yes, set the panel you want to launch automatically as the "Starter Panel". Then in settings, you can set the starter panel to automatically launch.  That panel will start as soon as the app is launched.


![image](https://user-images.githubusercontent.com/1140573/205461491-c1e3f72c-9ced-4bc2-853d-4edede87ab5a.png)


![image](https://user-images.githubusercontent.com/1140573/205461463-270d354a-938f-4fd2-8225-53b0621e672a.png)

**Reply by jcummings2** [reply]
Thanks, Dan. That will work great for a single panel.

Is there also a way to launch different panel either by name or from a file?

**Reply by dannagle** [reply]
Panels can transition to other panels using the "panel:X" syntax. 

See screenshot:
![image](https://user-images.githubusercontent.com/1140573/205462161-3e88a3d4-cd29-482a-877f-b178b472008d.png)

You can have your starter panel be a dashboard with buttons that you click to go to sub panels. You can have other commands inside the buttons that executes the panel:X command (like first do a setup sequence and then transition to sub panel).

**Comment by dannagle** [comment]
FYI... next version of Packet Sender will support launching the starter panel in a read-only mode and without the backend UI.  Pass the argument `--starterpanel` .

**Reply by eleliope** [reply]
Hi Dan, 
I'm eleliope, I just introduced myself in the discussion "Welcome to the packet sender forums".
Regarding the launching in panel only mode option I'm not being able to get it with the --starterpanel command from the command line. I'm not a programmer, so I'm sure I'm missing something.  

At the moment, I get a similar effect by configuring the program to open starter panel on startup, and to always open the main instance of the program minimised. But this way the panel window retains the ability to be edited, and it would be nice if it could be opened in read-only mode.

**Reply by dannagle** [reply]
Make sure you are launching using the `.exe` version with `packetsender.exe` (not `.com` version) to use panels. Perhaps that is what is throwing you off.


To help you out, Packet Sender comes bundled with a starter example panel if you have not configured one yet. 

See this screenshot using Powershell:
![image](https://github.com/dannagle/PacketSender/assets/1140573/85d5ab5f-ef45-4d05-85b1-9c68517256ba)


Here it is again using `cmd` in case you prefer that...

![image](https://github.com/dannagle/PacketSender/assets/1140573/5e605ff9-9ac1-4969-bca8-a2ced00c11c5)

**Reply by eleliope** [reply]
Indeed, that was the problem. I was using packetsender.com instead of packetsender.exe. Thank you very much, Dan, for your quick and clear reply.
